core/shared/src/main/scala/com/monovore/decline/Parser.scala

```diff
@@ -319,12 +319,14 @@ private[decline] object Parser {
           .getOrElse(Result.missingArgument)
     }
 
-    case class Subcommand[A](name: String, action: Parser[A]) extends Accumulator[A] {
+    case class Subcommand[A](name: String, action: Parser[A], env: Map[String, String]) extends Accumulator[A] {
```

Comment:
Ah, hadn't realized we'd need to deal with threading this through -- this looks good!